substrate/frame/identity/src/lib.rs

```diff
@@ -922,6 +961,129 @@ pub mod pallet {
 			});
 			Ok(())
 		}
+
+		/// Reap an identity, clearing associated storage items and refunding any deposits. Calls
+		/// the `ReapIdentityHandler`, which can be implemented in the runtime. This function is
+		/// very similar to (a) `clear_identity`, but called on a `target` account instead of self;
+		/// and (b) `kill_identity`, but without imposing a slash.
+		///
+		/// Parameters:
+		/// - `target`: The account for which to reap identity state.
+		///
+		/// Origin must be the `ReapOrigin`.
+		#[pallet::call_index(15)]
+		#[pallet::weight(T::WeightInfo::reap_identity(
+			T::MaxRegistrars::get(),
+			T::MaxSubAccounts::get()
+		))]
+		pub fn reap_identity(
```

Comment:
Because we have to "migrate" the deposits to a new runtime, and this pallet holds funds in reserve. To migrate that directly, it would mean minting funds in the genesis state of the new parachain, and burning funds in a Relay Chain runtime upgrade. IMO that is not very transparent; people should be able to see a transaction that moved their funds from one place to another.

That is the purpose of the `ReapIdentityHandler`: In the case of migrating to a parachain, it will teleport the funds needed for the deposit there. This will make the migration of funds transparent. It will also allow parachain deposit calculation to happen in the handler to ensure that the correct amount is teleported there.